slack/resource_conversation.go

```diff
@@ -107,6 +114,17 @@ func resourceSlackConversationCreate(ctx context.Context, d *schema.ResourceData
 	isPrivate := d.Get("is_private").(bool)
 
 	channel, err := client.CreateConversationContext(ctx, name, isPrivate)
+	if err != nil && err.Error() == "name_taken" && d.Get("adopt_existing_channel").(bool) {
+		channel, err = findExistingChannel(ctx, client, name, isPrivate)
+		if err == nil && channel.IsArchived {
+			// ensure unarchived first if adopting existing channel, else other calls below will fail
+			if err := client.UnArchiveConversationContext(ctx, channel.ID); err != nil {
```

Comment:
if the channel is currently archived, even if it is configured to eventually be archived, we need to unarchive it as many of the calls below will fail if called on an archived channel.

Reply:
NOTE! this api does not work with bot tokens (despite documentation to the contrary). The provider _must_ be configured with a user token.